### DISCUSSION TITLE
Add parameter support to Email fields

### DISCUSSION THREAD

**Discussion by alexjcollins** [verbb/hyper#181] [Ideas]
At the moment, if you put a parameter in the Email field (i.e. test@test.com?subject=hi), you get a validation error.

I'm guessing the solution is to use the built-in `URL Suffix` field to append any parameters to the email.

That said, it would be useful to support the parameters, as a number of our clients use [tools like this](https://mailtolink.me) to build their links. It would save them from having to slice the resulting link into two parts.

**Comment by engram-design** [comment]
That's a good point, we use the [`yii\validators\EmailValidator`](https://github.com/yiisoft/yii2/blob/master/framework/validators/EmailValidator.php) to validate the email value.

I'm kind of hesitant to change the validation rules to cater for query strings. I know they're valid for your use-case, but the value you've provided technically isn't an email. It's a mailto link, which supports all those extra parameters. I'm just thinking symantics, and all other use-cases people use where they expect the value of the field to _just_ be a valid email address.

I'm thinking you _probably_ want to steer your clients to using a "Custom" link type, which is much more flexible, and allows for anything to be added. This should suit if your clients are using that utility (that's a great resource, didn't know about that) to generate links like:

```
mailto:test@test.com?subject=Hey%20there&body=Just%20some%20text
```

They can just copy/paste that into the link content. The only benefit to using an Email link type is the cognitive load of saying "this is an email link, so I'll pick email" and that it'll add the `mailto:` part for you. But as it stands, I expect your clients would need to remove that in the string when they enter content.

So yep the solution at the moment is to either use the URL Suffix field, or use a Custom link type. Or, finally - you could extend the Email link type yourself and remove this validation rule (or tweak it). I think I'll stick with that for now...

**Comment by alexjcollins** [comment]
Thanks for the quick and thorough response, @engram-design.

That is a good point about using the Custom link type, i'll probably suggest that. If I get pushback, I'll look into extending the Email link type.

Cheers!
